packages/frint-cli/commands/init.js

```diff
@@ -37,17 +39,27 @@ module.exports = createApp({
           const example = deps.params.example || 'counter';
           const dir = deps.pwd;
 
-          const cmds = [
-            `mkdir -p ${dir}`,
-            `curl https://codeload.github.com/Travix-International/frint/tar.gz/master | tar -xz -C ${dir} --strip=3 frint-master/examples/${example}`,
-          ];
+          function streamFrintExampleToDir() {
+            request('https://codeload.github.com/Travix-International/frint/tar.gz/master')
+              .on('error', deps.console.error)
+              .pipe(tar.x({
+                filter: path => path.indexOf(`frint-master/examples/${example}/`) === 0,
+                strip: 3,
+                C: dir
+              }))
+              .on('error', deps.console.error)
+              .on('finish', () => deps.console.log('Done!'));
+          }
 
           deps.console.log('Initializing...');
-          const cmdPromises = cmds.map(cmd => exec.shell(cmd));
 
-          Promise.all(cmdPromises)
-            .then(() => deps.console.log('Done!'))
-            .catch(e => deps.console.error(e));
+          mkdirp(dir, function (error) {
+            if (error) {
+              deps.console.error(error);
+              return;
+            }
+            streamFrintExampleToDir();
```

Comment:
shouldn't it be synchronized at some point using Promises?

Reply:
Afaik, no synchronization is required. The code is executed in a fire and forget manner, but the runtime ensures in this case, that it's fully executed before the process is terminated. Should behave similarly to promises.